addons/website/controllers/main.py

```diff
@@ -222,7 +222,12 @@ def robots(self, **kwargs):
         # Don't use `request.website.domain` here, the template is in charge of
         # detecting if the current URL is the domain one and add a `Disallow: /`
         # if it's not the case to prevent the crawler to continue.
-        return request.render('website.robots', {'url_root': request.httprequest.url_root}, mimetype='text/plain')
+        content = request.env['ir.ui.view']._render_template(
+            'website.robots', {'url_root': request.httprequest.url_root}
+            )
+        extra_rules = self._get_additional_robots_rules_web()
+        content += '\n' + extra_rules
```

Comment:
Seems better to me to pass additional routes to the render context, and then add them in qweb.
Which would also avoid having to turn a lazy call into a non-lazy one.

Reply:
Added the suggestions, thanks for the review 🙂